### ISSUE TITLE
About forward

### ISSUE THREAD

**Issue by MontaEllis** [tomguluson92/PRNet_PyTorch#2]
I have trained a model in Pytorch with your code. But I wonder how to transform the Pytorch model to tensorflow one. Cause your code don't have forward code.
I hope you can help me. Thanks a lot.

**Comment by tomguluson92** [comment]
Hi, I have inference stage finished and upload my test model `results/latest.pth`, you can test it!

However, I found there may be some mistakes in the original `face3d` repo. I will strive to fix that in the coming days.

**Comment by kongyulian99** [comment]
> Hi, I have inference stage finished and upload my test model results/latest.pth, you can test it!
> However, I found there may be some mistakes in the original face3d repo. I will strive to fix that in the coming days.

The author don't say how to normalize pos_map label, when the output of resfcn256 is in [0, 1] range and *.npy have some negative elements.

**Comment by tomguluson92** [comment]
> > Hi, I have inference stage finished and upload my test model results/latest.pth, you can test it!
> > However, I found there may be some mistakes in the original face3d repo. I will strive to fix that in the coming days.
> 
> The author don't say how to normalize pos_map label, when the output of resfcn256 is in [0, 1] range and *.npy have some negative elements.

Hi, I just follow the process as same as the original repo. Do you mean that there may be some problems about normalization in face3d ?

**Comment by kongyulian99** [comment]
> > > Hi, I have inference stage finished and upload my test model results/latest.pth, you can test it!
> > > However, I found there may be some mistakes in the original face3d repo. I will strive to fix that in the coming days.
> > 
> > 
> > The author don't say how to normalize pos_map label, when the output of resfcn256 is in [0, 1] range and *.npy have some negative elements.
> 
> Hi, I just follow the process as same as the original repo. Do you mean that there may be some problems about normalization in face3d ?

I think problems are in traning part, because *.npy labels have some negative elements, but the output of resfcn256 is Sigmoid which output range is [0, 1]. Your labels: 
![image](https://user-images.githubusercontent.com/28080803/68917119-d7372300-07a3-11ea-80f3-cff6af4105eb.png) only devided by 255.0? But I don't know how to normalize the labels by the way @@!

**Comment by tomguluson92** [comment]
Hi, I think a `np.clip(uv_map, 0, 1)` should be added here to fix this problem?

**Comment by kongyulian99** [comment]
![image](https://user-images.githubusercontent.com/28080803/69206999-5b651e00-0b89-11ea-8c74-ef64d3b82cc4.png)
This is my final results.

**Comment by tomguluson92** [comment]
> ![image](https://user-images.githubusercontent.com/28080803/69206999-5b651e00-0b89-11ea-8c74-ef64d3b82cc4.png)
> This is my final results.

Looks good to me, good job~

**Comment by MontaEllis** [comment]
> 
> 
> ![image](https://user-images.githubusercontent.com/28080803/69206999-5b651e00-0b89-11ea-8c74-ef64d3b82cc4.png)
> This is my final results.

Hi, I want to know how you can get such a good result. I can train and forward, but the result is poor.

**Comment by tomguluson92** [comment]
> > ![image](https://user-images.githubusercontent.com/28080803/69206999-5b651e00-0b89-11ea-8c74-ef64d3b82cc4.png)
> > This is my final results.
> 
> Hi, I want to know how you can get such a good result. I can train and forward, but the result is poor.

Based on my own view, you simply need more training data to make your network more robust. `300WLP` provided by me is not enough for a precise high-varied viewpoint adaptive model.
